express/parsers/apps/espresso/parser.py

```diff
@@ -356,14 +356,30 @@ def _is_sternheimer_gw_calculation(self):
         """
         return False
 
+    @property
     def xml_parser(self):
         if self._xml_parser is None:
-            self._xml_parser = Espresso640XMLParser(self.find_xml_file())
+            xml_file = self.find_xml_file()
+            self._xml_parser = Espresso640XMLParser(xml_file)
         return self._xml_parser
 
-
-    def fermi_energy(self):
-        return self.ase_parser.fermi_energy()
-
-    def eigenvalues_at_kpoints(self) -> List:
-        return self.ase_parser.eigenvalues_at_kpoints()
+    def find_xml_file(self, cwd=None):
+        if cwd is None:
+            cwd = self.work_dir
+        files = []
+        directories = []
+        for item in map(lambda i: os.path.join(cwd, i), os.listdir(cwd)):
+            if os.path.isdir(item):
+                directories.append(item)
+            elif os.path.isfile(item):
+                files.append(item)
+
+        for file in files:
+            if file.endswith("xml"):
```

Comment:
1. This one will only ever find one XML file - is there a specific naming convention for it? This coulbe be prone to error if a workflow has multiple steps reusing the same directory. 

2. How did `find_xml_file()` work before - why did we need to adjust this?

Reply:
Figured out the failure! What was going on is that the settings.XML_DATA_FILE variable is set to `data-file.xml`. The new versions of Espresso store this as `data-file-schema.xml`. Fixed this by storing it as an attribute of the different parsers, for now.